.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,12 @@
+### Description
+
+Please explain the changes you made, including a reference to the related issue if applicable
+
+### Checklist
+- [ ] Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
+- [ ] Created tests, which fail without this PR, if possible [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
+- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
+- [ ] Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)
+- [ ] Include a screenshot if changing a GUI
```

Comment:
I feel this last item should definitely be moved up in the list.
If I had to put them in order, I would go

- format (hard requirement)
- screenshot (soft requirement - we *always* ask for that anyway)
- tutorials/documentation (*have* to be adjusted to account for changes, *could* be improved if something's not documented yet)
- new tests (not *mandatory* but encouraged)
- cherry-pick to other branches? (this is mostly a maintainer decision, but suggestions by external requestors are welcome)

I would actually include the importance of the various items in parenthesis, I don't think it makes sense to ask requestors to check all of these items for every request. IMHO, this raises the barrier to submit (reasonably simple) patches far too much for new contributors.